components/brave_ads/rust/lib.rs

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use lazy_static::lazy_static;
+use regex::Regex;
+
+#[cxx::bridge(namespace = brave_ads)]
+mod ffi {
+    extern "Rust" {
+        fn strip_non_alpha_character(text: &str) -> String;
+        fn strip_non_alpha_numeric_character(text: &str) -> String;
+        fn parse_and_sanitize_og_tag_attribute(html: &str) -> String;
+        fn parse_html_og_tag_attribute(html: &str) -> String;
+        fn collapse_whitespace(text: &str) -> String;
+    }
+}
+
+fn strip_non_alpha_character(text: &str) -> String {
+    fn get_pattern() -> String {
+        let escaped_characters = regex::escape("!\"#$%&'()*+,-./:<=>?@\\[]^_`{|}~");
+        format!(
+            "[[:cntrl:]]|\\\\(t|n|v|f|r)|[\\t\\n\\v\\f\\r]|\\\\x[[:xdigit:]][[:xdigit:]]|[{}]|\\S*\\d+\\S*",
+            escaped_characters
+        )
+    }
+
+    lazy_static! {
+        static ref CHAR_REGEX: Regex = Regex::new(&get_pattern()).unwrap();
+    }
+    let output = CHAR_REGEX.replace_all(text, " ");
+
+    collapse_whitespace(&output)
+}
+
+fn strip_non_alpha_numeric_character(text: &str) -> String {
+    fn get_pattern() -> String {
+        let escaped_characters = regex::escape("!\"#$%&'()*+,-./:<=>?@\\[]^_`{|}~");
+        format!(
+            "[[:cntrl:]]|\\\\(t|n|v|f|r)|[\\t\\n\\v\\f\\r]|\\\\x[[:xdigit:]][[:xdigit:]]|[{}]",
+            escaped_characters
+        )
+    }
+
+    lazy_static! {
+        static ref CHAR_REGEX: Regex = Regex::new(&get_pattern()).unwrap();
+    }
+    let output = CHAR_REGEX.replace_all(text, " ");
+
+    collapse_whitespace(&output)
+}
+
+fn parse_and_sanitize_og_tag_attribute(html: &str) -> String {
+    let tag_attribute = parse_html_og_tag_attribute(html);
+    let sanitized_tag_attribute = strip_non_alpha_character(&tag_attribute);
+    sanitized_tag_attribute.to_lowercase()
+}
+
+fn parse_html_og_tag_attribute(html: &str) -> String {
+    lazy_static! {
+        static ref TAG_REGEX: Regex =
+            Regex::new("(<[^>]*\\bproperty=[\"']og:title[\"'][^<]*>)").unwrap();
```

Comment:
@ptjames, embedding regular expressions were changed to more strict versions.

Reply:
Additional check for `property=` looks good to me

Reply:
This should work fine for our case because this is actually not a raw HTML, but the output from `XmlSerializer().serializeToString()` call.